File: 1390-four-divisors/1390-four-divisors.cpp
class Solution {
public:
    int sumFourDivisors(vector<int>& nums) {
        int res = 0;
        for(int i = 0; i < nums.size(); i++)
        {
            int tot = nums[i] + 1;
            int size = 2;
            for(int j = 2; j <= sqrt(nums[i]); j++)
            {
                if(nums[i] % j == 0)
                {
                    tot += j + (nums[i] / j);
                    if(j == (nums[i] / j))
                    {
                        size++;
                    }
                    else
                    {
                        size += 2;
                    }
                }
            }
            if(size == 4)
            {
                res += tot;
            }
        }
        return res;
    }
};

File: 1390-four-divisors/1390-four-divisors.py
class Solution:
    def sumFourDivisors(self, nums: List[int]) -> int:
        res = 0
        for num in nums:
            tot = num + 1
            size = 2
            for i in range(2, int(sqrt(num) + 1)):
                if num % i == 0:
                    tot += i + (num // i)
                    if i == (num // i):
                        size += 1
                    else:
                        size += 2
            if size == 4:
                res += tot
        return res